src/global_mapper_ros/global_mapper_ros.cc

```diff
@@ -2,77 +2,175 @@
 }
 
 void GlobalMapperRos::InitSubscribers() {
   point_cloud_sub_ = nh_.subscribe<pcl::PointCloud<pcl::PointXYZ> >("cloud_topic", 10, &GlobalMapperRos::PointCloudCallback, this);
 }
 
 void GlobalMapperRos::InitPublishers() {
-  map_pub_ = pnh_.advertise<PointCloud>("map_topic", 1);
-  map_pub_timer_ = nh_.createTimer(ros::Duration(1.0), &GlobalMapperRos::PublishMap, this);
+  map_pub_ = nh_.advertise<visualization_msgs::MarkerArray>("map_topic", 1);
+  if (publish_voxel_map_) {
+    map_pub_timer_ = nh_.createTimer(ros::Duration(1.0), &GlobalMapperRos::PublishMap, this);
+  }
+}
+
+/*
+  Return a RGB colour value given a scalar v in the range [vmin,vmax]
+  In this case each colour component ranges from 0 (no contribution) to
+  1 (fully saturated). The colour is clipped at the end of the scales if v is outside
+  the range [vmin,vmax]
+*/
+std::vector<double> GlobalMapperRos::GrayscaleToRGBJet(double v, double vmin, double vmax) {
+  std::vector<double> c(3, 1.0);  // white
+  double dv;
+
+  if (v < vmin)
+    v = vmin;
+  if (v > vmax)
+    v = vmax;
+  dv = vmax - vmin;
+
+  if (v < (vmin + 0.25 * dv)) {
+    c[0] = 0;
+    c[1] = 4 * (v - vmin) / dv;
+  } else if (v < (vmin + 0.5 * dv)) {
+    c[0] = 0;
+    c[2] = 1 + 4 * (vmin + 0.25 * dv - v) / dv;
+  } else if (v < (vmin + 0.75 * dv)) {
+    c[0] = 4 * (v - vmin - 0.5 * dv) / dv;
+    c[2] = 0;
+  } else {
+    c[1] = 1 + 4 * (vmin + 0.75 * dv - v) / dv;
+    c[2] = 0;
+  }
+
+  return c;
 }
 
 void GlobalMapperRos::PublishMap(const ros::TimerEvent& event) {
   // lock
-  std::lock_guard<std::mutex> lock(global_mapper_.map_mutex_);
-
-  // populate
-  PointCloud map;
-  map.header.frame_id = "world";
-  map.height = 1;
-
-  int ixyz[3] = {0};
-  int count = 0;
-  for (int i = 0; i < global_mapper_.global_map_.size(); i++) {
-    if (global_mapper_.global_map_[i] > 0.0) {
-      count++;
-      global_mapper_.IndToCoord(i, ixyz);
-      map.points.push_back(pcl::PointXYZ(ixyz[0], ixyz[1], ixyz[2]));
+  std::lock_guard<std::mutex> lock(global_mapper_ptr_->map_mutex_);
+
+  // get occuppied voxels
+  std::vector<std::vector<double> > voxel_vec;
+
+  double xyz[3] = {0.0};
+  std::vector<double> voxel(3, 0.0);
+  for (int i = 0; i < global_mapper_ptr_->voxel_map_ptr_->num_cells; i++) {
+    global_mapper_ptr_->voxel_map_ptr_->indToLoc(i, xyz);
+
+    if (global_mapper_ptr_->voxel_map_ptr_->readValue(xyz) > 0.0) {
+      voxel[0] = xyz[0];
+      voxel[1] = xyz[1];
+      voxel[2] = xyz[2];
+      voxel_vec.push_back(voxel);
     }
   }
-  map.width = map.points.size();
-  ROS_INFO("map points: %i", count);
+  int num_voxels = voxel_vec.size();
+  ROS_INFO("num_voxels: %u", num_voxels);
+
+  visualization_msgs::MarkerArray marker_array;
+
+  visualization_msgs::Marker marker;
+  std::vector<double> rgb(3, 1.0);
+  for (int i = 0; i < num_voxels; ++i) {
+    // create voxel marker
+    marker.header.frame_id = "world";
+    marker.header.stamp = ros::Time::now();
+    marker.id = i;
```

Comment:
Nope.  That is just the ID of the Marker being added to this MarkerArray with a total of num_voxels Markers.